Assets/Mirage/Runtime/NetworkSceneManager.cs

```diff
@@ -182,210 +173,468 @@ internal void OnClientChangeScene(string scenePath, SceneOperation sceneOperatio
+        /// <param name="sceneOperation">Choose type of scene loading we are doing <see cref="SceneOperation"/>.</param>
+        private void ServerSceneLoading(string scenePath, INetworkPlayer[] players,
+            bool shouldClientLoadOrUnloadNormally, SceneOperation sceneOperation = SceneOperation.Normal)
         {
-            logger.Log("NetworkSceneManager.OnServerAuthenticated");
+            if (string.IsNullOrEmpty(scenePath))
+            {
+                throw new ArgumentNullException(nameof(scenePath),
+                    "[NetworkSceneManager] - ServerChangeScene: " + nameof(scenePath) + " cannot be empty or null");
+            }
 
-            player.Send(new SceneMessage { scenePath = ActiveScenePath, additiveScenes = additiveSceneList.ToArray() });
-            player.Send(new SceneReadyMessage());
+            //if(players == null || players.Length == 0)
+            //    throw new ArgumentNullException(nameof(scenePath),
+            //        "[NetworkSceneManager] - list of player's cannot be null or no players.");
+
+
+            if (logger.LogEnabled()) logger.Log("[NetworkSceneManager] - ServerChangeScene " + scenePath);
+
+            // Let server prepare for scene change
+            logger.Log("[NetworkSceneManager] - OnServerChangeScene");
+
+            ServerStartedSceneChange?.Invoke(scenePath, sceneOperation);
+
+            if (!Server.LocalClientActive)
+                LoadSceneAsync(scenePath, players, sceneOperation).Forget();
+
+            // notify all clients about the new scene
+            if (players.Length > 0 && shouldClientLoadOrUnloadNormally)
+                NetworkServer.SendToMany(players,
+                    new SceneMessage {MainActivateScene = scenePath, SceneOperation = SceneOperation.Normal});
+            else
+            {
+                NetworkServer.SendToMany(players,
+                    new SceneMessage {MainActivateScene = scenePath, SceneOperation = sceneOperation});
+            }
         }
 
         /// <summary>
-        /// This causes the server to switch scenes and sets the ActiveScenePath.
-        /// <para>Clients that connect to this server will automatically switch to this scene. This automatically sets clients to be not-ready. The clients must call Ready() again to participate in the new scene.</para>
+        /// 
         /// </summary>
-        /// <param name="scenePath"></param>
-        /// <param name="operation"></param>
-        /// <exception cref="ArgumentNullException">Scene path was not valid.</exception>
-        public void ChangeServerScene(string scenePath, SceneOperation sceneOperation = SceneOperation.Normal)
+        /// <param name="scene"></param>
+        /// <param name="players"></param>
+        /// <param name="sceneOperation"></param>
+        private void ServerSceneUnLoading(Scene scene, INetworkPlayer[] players, SceneOperation sceneOperation)
```

Comment:
do we need sceneOperation here, isn't this always unload?

Reply:
Good catch